### PR TITLE
ft: ZENKO-561: bucket attributes handling fixed

### PR DESCRIPTION
**Pull request by ploki**
on putBucketAttributes and getBucketAttributes

### CODE REVIEW

**Review by JianqinWang — DISMISSED** [review]

**Comment by ploki** [comment]
@bert-e wait

**Review by alexanderchan-scality — DISMISSED** [review]

**Review by nicolas2bert — DISMISSED** [review]

File: lib/storage/metadata/proxy/BucketdRoutes.js [hunk 1]
```diff
@@ -32,15 +32,19 @@ class BucketdRoutes {
...
-            bucketName, logger, (err, data) =>
-                                sendResponse(req, res, logger, err,
-                                    BucketInfo.fromObj(data).serialize())
-                            );
+            bucketName, logger, (err, data) => {
+                if (data === undefined) {
+                    return sendResponse(req, res, logger,
+                                        errors.NoSuchBucket);
+                }
+                return sendResponse(req, res, logger, err,
```
Comment (rahulreddy) [inline lib/storage/metadata/proxy/BucketdRoutes.js]: If there was a TCP error making a request, would data be `undefined` or `null` and `err` be set?

**Review by rahulreddy — CHANGES_REQUESTED** [review]

File: lib/storage/metadata/proxy/BucketdRoutes.js (hunk 1 above)
Comment (ploki) [reply lib/storage/metadata/proxy/BucketdRoutes.js]: err and data come from the MetadataWrapper and are not supposed to carry information about the underlying TCP connection, if any.
But I might want to check for an error before checking the payload validity

**Review by rahulreddy — DISMISSED** [review]

File: lib/storage/metadata/proxy/BucketdRoutes.js (hunk 1 above)
Comment (rahulreddy) [reply lib/storage/metadata/proxy/BucketdRoutes.js]: Since we follow error first callbacks style, it's inline.

**Review by bennettbuchanan — APPROVED** [review]

**Review by rahulreddy — APPROVED** [review]

**Pull request merged**